plugins/soundsourcem4a/audiosourcem4a.cpp

```diff
@@ -0,0 +1,307 @@
+#include "audiosourcem4a.h"
+
+#include "sampleutil.h"
+
+#ifdef __WINDOWS__
+#include <io.h>
+#include <fcntl.h>
+#endif
+
+#ifdef _MSC_VER
+#define S_ISDIR(mode) (mode & _S_IFDIR)
+#define strcasecmp stricmp
+#define strncasecmp strnicmp
+#endif
+
+#ifdef __M4AHACK__
+typedef uint32_t SAMPLERATE_TYPE;
+#else
+typedef unsigned long SAMPLERATE_TYPE;
+#endif
+
+namespace Mixxx {
+
+namespace
+{
+
+// AAC: "... each block decodes to 1024 time-domain samples."
+const AudioSource::size_type kFramesPerSampleBlock = 1024;
+
+// MP4SampleId is 1-based
+const MP4SampleId kMinSampleId = 1;
+
+// Decoding will be restarted one or more blocks of samples
+// before the actual position to avoid audible glitches.
+// Two blocks of samples seems to be enough here.
+const MP4SampleId kSampleIdPrefetchCount = 2;
+
+MP4TrackId findAacTrackId(MP4FileHandle hFile) {
```

Comment:
My arguments for a free function in an anonymous namespace are:
- This function does not need access to private/protected parts of any class. There is no need to make it a member function of a class even if it would be "static".
- The anonymous namespace makes it invisible for the linker out of this compilation unit and avoids naming conflicts.
- Keeping header files short and and class definitions clean from implementation details improves readability and reduces compilation times.

Well, if my arguments still don't convince you I will change it ;) I just wanted to explain why I did it this way.